app/models/cms/file.rb

```diff
@@ -1,4 +1,7 @@
+# frozen_string_literal: true
+
 class CMS::File < ApplicationRecord
+  include ThreeScale::Search::Scopes
```

Comment:
I think this is only for full-text searching, since we don't do this, perhaps remove this include?

Reply:
We need that module to be able to call `scope_search` which is what  I'm using to filter the results. It expects some scopes to be created in the models. Not sure how this can be used for full-text searching.

Reply:
https://github.com/3scale/porta/blob/66355ab6fe8679fa7165caa6da326be9a60fbd14/app/lib/three_scale/search.rb#L117

To me it sounds like sphinx specific. Do you see any added benefits of using that instead of calling the scopes directly?

Or in case the call is complicated, maybe it is even better to have a `CMSFileParentSearchService`.

Reply:
Yes, I don't know why the word `sphinx` appears there, the code doesn't seem to be related. If you look at that method, what it does is to infer the list of scopes to be called based on the received parameters, and call the scopes. No Sphinx involved apparently. That's the main advantage of this module, to avoid calling the scopes manually, since depending on the received parameters a different list of scopes should be called, and we want to avoid implementing such logic in every controller.

Reply:
My quick, uneducated and non-thorough tracking this method, shows me `params[:query] = escape_query(params)` and then inside `#escape_query` we have `ThinkingSphinx::Query.escape(query)`. So to me it looks like a way to search with sphinx query within some scopes somehow easier.

While also it seems to be usable without a sphinx query, where the `:query` parameter would be `nil`. It still seems to me like a sphinx specific thing.

While you might be right that intention was not to be used only by full text searching, I'm not convinced yet. And need to understand better how this works and other usages within porta.

In the likely case you already did so, we may have a chat to get this sorted out easier.

Reply:
I took a look at this. It works this way: the module `ThreeScale::Search::Scopes` allows you to bind params to scopes. It expects you to add a `by_param` scope to the model, for each `param`. The list of accepted scopes is set by `allowed_search_scopes` at the model. For each key in that list, you must add the scope. See [an example](https://github.com/3scale/porta/pull/3286/files#diff-b71d8856f040113f12c49a001e8bc4f45ffd147c1d34d27ca5eb281d917d5b21R14-R18).

Additionally, it allows you to create a scope called `by_query`, which will be called with the `query` param escaped. You are supposed to call Sphinx from that scope. We have a module called [`Searchable`](https://github.com/3scale/porta/blob/013be400daa96cfef89cc5f8d5cdd28b57f6403a/app/concerns/searchable.rb) which implements the `by_query` scope to call Sphinx. That module includes `ThreeScale::Search::Scopes`. So the idea is that you include `Searchable` from a model and you have a full scoped + sphinx search for that model.

In this PR, we are not including `Searchable` nor we are implementing `by_query` in any model, the `query` param is ignored and Sphinx is not called. In fact, when https://github.com/3scale/porta/pull/3296 is merged the `query` param will be rejected.

I think this is safe to merge.

Reply:
@jlledom , thank you. Now it makes more sense. I was actually thinking about the [Searchable](https://github.com/3scale/porta/blob/013be400daa96cfef89cc5f8d5cdd28b57f6403a/app/concerns/searchable.rb) scope but didn't look at the things enough to realize my confusion.

I'm still not fully convinced about the usefulness of this approach but I'm ok to use this approach. Can we remove this `sphinx_search` alias though? It is also confusing. And add explanation about this method as a comment above it?

Reply:
I removed the `sphinx_search` alias.

> I'm still not fully convinced about the usefulness of this approach

What's your idea?